src/Body.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Http\Client;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Allows a more input types for body data
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+interface Body
+{
+    /**
+     * Returns a set of headers which is needed to correctly send the body
+     *
+     * Note: these headers SHOULD be overwritten if any matching header is passed manually
```

Comment:
Well, yes, probably this notice should not be here. The point is that headers returned from a body should be overwritten by the headers manually passed to the client. Just to allow full control for the user.